### PR TITLE
Added prefix length to ipam

### PR DESCRIPTION
**Pull request by IstrateM**

### CODE REVIEW

File: pkg/nseconfig/config.go [hunk 1]
```diff
@@ -37,8 +37,9 @@ type VL3 struct {
 }
 
 type IPAM struct {
-	PrefixPool string
-	Routes     []string
+	PrefixPool   string
+	PrefixLength string
```
Comment (danielvladco) [inline pkg/nseconfig/config.go]: why not make PrefixLength pf type int?

File: pkg/nseconfig/validate.go [hunk 2]
```diff
@@ -46,6 +47,14 @@ func (v VL3) validate() error {
 		}
 	}
 
+	if pl, err := strconv.Atoi(v.IPAM.PrefixLength); err == nil {
```
Comment (danielvladco) [inline pkg/nseconfig/validate.go]: after prefixlength will be int this will be missing, also you should check if `err != nil` almost all the time

Comment (IstrateM) [reply pkg/nseconfig/validate.go]: done

File: pkg/nseconfig/config.go (hunk 1 above)
Comment (IstrateM) [reply pkg/nseconfig/config.go]: done

File: k8s/helm/vl3/values.yaml [hunk 3]
```diff
@@ -18,7 +18,8 @@ nsm:
   serviceName: vl3-service
 
 ipam:
-  prefixPool: "172.31.0.0/16"
+  prefixPool: "172.31.0.0/24"
```
Comment (danielvladco) [inline k8s/helm/vl3/values.yaml]: change to defaultPrefixPool

File: k8s/helm/vl3/values.yaml [hunk 4]
```diff
@@ -18,7 +18,8 @@ nsm:
...
+  prefixLength: "24"
   # uniqueOctet: 
 
 replicaCount: 1
```
Comment (danielvladco) [inline k8s/helm/vl3/values.yaml]: new line

File: k8s/helm/vl3/templates/vl3-nse-config.yaml [hunk 5]
```diff
@@ -15,5 +15,6 @@ data:
       vl3:
        ipam:
           prefixpool: {{ .Values.ipam.prefixPool | quote }}
```
Comment (danielvladco) [inline k8s/helm/vl3/templates/vl3-nse-config.yaml]: change to defaultPrefixPool

File: k8s/helm/vl3/templates/vl3-nse-config.yaml [hunk 6]
```diff
@@ -15,5 +15,6 @@ data:
...
+          prefixlength: {{ .Values.ipam.prefixLength | quote }}
```
Comment (danielvladco) [inline k8s/helm/vl3/templates/vl3-nse-config.yaml]: change the rest of the fields ad you are good to go

File: k8s/helm/vl3/values.yaml (hunk 3 above)
Comment (IstrateM) [reply k8s/helm/vl3/values.yaml]: done

File: k8s/helm/vl3/values.yaml (hunk 4 above)
Comment (IstrateM) [reply k8s/helm/vl3/values.yaml]: done

File: k8s/helm/vl3/templates/vl3-nse-config.yaml (hunk 5 above)
Comment (IstrateM) [reply k8s/helm/vl3/templates/vl3-nse-config.yaml]: done

File: k8s/helm/vl3/templates/vl3-nse-config.yaml (hunk 6 above)
Comment (IstrateM) [reply k8s/helm/vl3/templates/vl3-nse-config.yaml]: done

**Review by danielvladco — APPROVED** [review]

**Pull request merged**